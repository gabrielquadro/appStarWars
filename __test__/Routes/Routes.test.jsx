import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Routes from '../../src/routes';

// Mocking dependencies
jest.mock('@expo/vector-icons', () => ({
  Feather: 'FeatherMock',
}));

jest.mock('react-native-gesture-handler', () => {
  const View = require('react-native/Libraries/Components/View/View');
  return {
    GestureHandlerModule: {
      attachGestureHandler: jest.fn(),
      createGestureHandler: jest.fn(),
      dropGestureHandler: jest.fn(),
      updateGestureHandler: jest.fn(),
      ...(jest.requireActual('react-native-gesture-handler').GestureHandlerModule || {}),
    },
    GestureHandlerRootView: View,
    gestureHandlerRootHOC: jest.fn(),
  };
});

describe('Routes', () => {
    //rendr home
  it('renders Home tab correctly', () => {
    const { toJSON } = render(<Routes />);
    expect(toJSON()).toMatchSnapshot();
  });

  //render favorite
  it('renders Favorites tab correctly', () => {
    const { getByText } = render(<Routes />);

    //Navigation to Favorites tab
    fireEvent.press(getByText('Favorites'));

    //verify if favorite screen is rendered
    const favoritesHeader = getByText('Favorites');
    expect(favoritesHeader).toBeDefined();
  });
});
