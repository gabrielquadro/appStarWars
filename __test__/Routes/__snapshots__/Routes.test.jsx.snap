// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes renders Home tab correctly 1`] = `
<View>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenNavigationContainer
      style={
        {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <RNSScreen
        activityState={2}
        collapsable={false}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        onGestureCancel={[Function]}
        sheetAllowedDetents="large"
        sheetCornerRadius={-1}
        sheetExpandsWhenScrolledToEdge={true}
        sheetGrabberVisible={false}
        sheetLargestUndimmedDetent="all"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            [
              {
                "backgroundColor": "rgb(242, 242, 242)",
                "flex": 1,
              },
              [
                {
                  "flex": 1,
                  "flexDirection": "column-reverse",
                },
                undefined,
              ],
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <RNSScreenStack
                style={
                  {
                    "flex": 1,
                  }
                }
              >
                <RNSScreen
                  collapsable={false}
                  gestureResponseDistance={
                    {
                      "bottom": -1,
                      "end": -1,
                      "start": -1,
                      "top": -1,
                    }
                  }
                  nativeBackButtonDismissalEnabled={false}
                  onAppear={[Function]}
                  onDisappear={[Function]}
                  onDismissed={[Function]}
                  onGestureCancel={[Function]}
                  onHeaderBackButtonClicked={[Function]}
                  onNativeDismissCancelled={[Function]}
                  onTransitionProgress={[Function]}
                  onWillDisappear={[Function]}
                  replaceAnimation="push"
                  sheetAllowedDetents="large"
                  sheetCornerRadius={-1}
                  sheetExpandsWhenScrolledToEdge={true}
                  sheetGrabberVisible={false}
                  sheetLargestUndimmedDetent="all"
                  stackPresentation="push"
                  style={
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                  swipeDirection="horizontal"
                >
                  <View
                    accessibilityElementsHidden={false}
                    importantForAccessibility="auto"
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        [
                          {
                            "flex": 1,
                          },
                          {
                            "backgroundColor": "rgb(242, 242, 242)",
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          {
                            "backgroundColor": "#000",
                            "flex": 1,
                            "paddingVertical": 10,
                          }
                        }
                        testID="home-screen"
                      >
                        <RCTScrollView
                          ListFooterComponent={[Function]}
                          data={[]}
                          getItem={[Function]}
                          getItemCount={[Function]}
                          keyExtractor={[Function]}
                          onContentSizeChange={[Function]}
                          onEndReached={[Function]}
                          onEndReachedThreshold={0.1}
                          onLayout={[Function]}
                          onMomentumScrollBegin={[Function]}
                          onMomentumScrollEnd={[Function]}
                          onScroll={[Function]}
                          onScrollBeginDrag={[Function]}
                          onScrollEndDrag={[Function]}
                          removeClippedSubviews={false}
                          renderItem={[Function]}
                          scrollEventThrottle={0.0001}
                          stickyHeaderIndices={[]}
                          testID="character-list"
                          viewabilityConfigCallbackPairs={[]}
                        >
                          <View>
                            <View
                              onLayout={[Function]}
                            >
                              <View
                                style={
                                  {
                                    "borderTopWidth": 1,
                                    "paddingVertical": 20,
                                  }
                                }
                                testID="loading-indicator"
                              >
                                <ActivityIndicator
                                  color="#FFD700"
                                  size="large"
                                />
                              </View>
                            </View>
                          </View>
                        </RCTScrollView>
                      </View>
                    </View>
                  </View>
                  <RNSScreenStackHeaderConfig
                    backButtonInCustomView={false}
                    backTitleVisible={true}
                    backgroundColor="#212121"
                    color="#FFD700"
                    direction="ltr"
                    disableBackButtonMenu={false}
                    hidden={false}
                    hideBackButton={false}
                    largeTitleHideShadow={false}
                    title="Character list"
                    titleColor="#FFD700"
                    titleFontWeight="bold"
                    topInsetEnabled={false}
                    translucent={false}
                  />
                </RNSScreen>
              </RNSScreenStack>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenNavigationContainer>
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="auto"
      style={
        {
          "backgroundColor": "#212121",
          "borderTopColor": "rgb(216, 216, 216)",
          "borderTopWidth": 0.5,
          "bottom": 0,
          "elevation": 8,
          "height": 49,
          "left": 0,
          "paddingBottom": 0,
          "paddingHorizontal": 0,
          "position": null,
          "right": 0,
          "transform": [
            {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <View
        accessibilityRole="tablist"
        style={
          {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityLabel="Home, tab, 1 of 2"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": true,
            }
          }
          accessibilityStates={
            [
              "selected",
            ]
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "flex": 1,
              },
              {
                "backgroundColor": "transparent",
              },
              {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  {
                    "opacity": 1,
                  },
                ]
              }
            >
              <FeatherMock
                color="#FFD700"
                name="home"
                size={25}
              />
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  {
                    "opacity": 0,
                  },
                ]
              }
            >
              <FeatherMock
                color="#FFFFFF"
                name="home"
                size={25}
              />
            </View>
          </View>
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "textAlign": "center",
                },
                {
                  "color": "#FFD700",
                },
                {
                  "fontSize": 10,
                },
                undefined,
              ]
            }
          >
            Home
          </Text>
        </View>
        <View
          accessibilityLabel="Favorites, tab, 2 of 2"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": false,
            }
          }
          accessibilityStates={[]}
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "flex": 1,
              },
              {
                "backgroundColor": "transparent",
              },
              {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  {
                    "opacity": 0,
                  },
                ]
              }
            >
              <FeatherMock
                color="#FFD700"
                name="star"
                size={25}
              />
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  {
                    "opacity": 1,
                  },
                ]
              }
            >
              <FeatherMock
                color="#FFFFFF"
                name="star"
                size={25}
              />
            </View>
          </View>
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "textAlign": "center",
                },
                {
                  "color": "#FFFFFF",
                },
                {
                  "fontSize": 10,
                },
                undefined,
              ]
            }
          >
            Favorites
          </Text>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
